pkg/scaffold/application.go

```diff
@@ -28,14 +29,35 @@ func NewApplications() (*Applications, error) {
 }
 
 func (apps *Applications) HelmValues(app string) (map[string]interface{}, error) {
-	var res map[string]interface{}
-	path := pathing.SanitizeFilepath(filepath.Join(apps.Root, app, "helm", app, "values.yaml"))
-	content, err := os.ReadFile(path)
+	res := make(map[string]interface{})
+	valuesFile := pathing.SanitizeFilepath(filepath.Join(apps.Root, app, "helm", app, "values.yaml"))
+	vals := make(map[string]interface{})
+	valsContent, err := os.ReadFile(valuesFile)
 	if err != nil {
 		return res, err
 	}
+	if err := yaml.Unmarshal(valsContent, &vals); err != nil {
+		return res, err
+	}
+
+	defaultValuesFile := pathing.SanitizeFilepath(filepath.Join(apps.Root, app, "helm", app, "default-values.yaml"))
+	defaultVals := make(map[string]interface{})
+	if utils.Exists(defaultValuesFile) {
+		defaultValsContent, err := os.ReadFile(defaultValuesFile)
+		if err != nil {
+			return res, err
+		}
+		if err := yaml.Unmarshal(defaultValsContent, &defaultVals); err != nil {
+			return res, err
+		}
+	}
 
-	err = yaml.Unmarshal(content, &res)
+	for k, v := range defaultVals {
+		res[k] = v
+	}
+	for k, v := range vals {
```

Comment:
isn't this merge technically too strict?  Like it wouldn't handle nested objects properly

Reply:
it's not the typical merge. We have to override default values with values

Reply:
Yeah I wasn't happy about that merge, but got caught up removing empty maps properly. Lukasz change should have fixed this merging.

@zreigz Wouldn't what we do in https://github.com/pluralsh/plural-cli/blob/main/pkg/scaffold/helm.go#L200 also be too strict then?

Reply:
@DavidSpek  here is another problem, I will make another PR for this